homeassistant/components/ptvsd/__init__.py

```diff
@@ -0,0 +1,66 @@
+"""
+Enable ptvsd debugger to attach to HA.
+
+Attach ptvsd debugger by default to port 5678.
+"""
+
+import logging
+from threading import Thread
+from asyncio import Event
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+DOMAIN = 'ptvsd'
+
+CONF_WAIT = 'wait'
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(
+            CONF_HOST, default='0.0.0.0'
+        ): cv.string,
+        vol.Optional(
+            CONF_PORT, default=5678
+        ): cv.port,
+        vol.Optional(
+            CONF_WAIT, default=False
+        ): cv.boolean,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Set up ptvsd debugger."""
+    import ptvsd
+
+    conf = config.get(DOMAIN)
+    if not conf:
+        return True
+
+    host = conf.get(CONF_HOST)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
OK